src/Shared/Compat/TypeExtensions.cs

```diff
@@ -8,5 +12,622 @@ public static bool IsEquivalentTo(this Type type, Type other)
         {
             return type.Equals(other);
         }
+
+        public static object InvokeMember(this Type type, string name, BindingFlags bindingFlags, object target, object[] providedArgs,
+            ParameterModifier[] modifiers, CultureInfo culture, string[] namedParams)
+        {
+            return new InvokeMemberHelper(type).InvokeMember(name, bindingFlags, target, providedArgs, modifiers, culture, namedParams);
+        }
+
+
+        class InvokeMemberHelper
```

Comment:
I assume/hope you didn't write all of this just now :smile: Comment source so we can have hope of removing it later or at least keeping it in sync?